### PR TITLE
save reorganization processed handle in meta for later resuming if crashed

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut  @qiuyesuifeng

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -150,22 +148,15 @@ func (d *ddl) onAddColumn(t *meta.Meta, job *model.Job) error {
...
-		if job.SnapshotVer == 0 {
-			var ver kv.Version
-			ver, err = d.store.CurrentVersion()
-			if err != nil {
-				return errors.Trace(err)
-			}
-
-			job.SnapshotVer = ver.Ver
+		var reorgInfo *reorgInfo
+		reorgInfo, err = d.getReorgInfo(t, job)
```
Comment (ngaut) [inline ddl/column.go]: reorgInfo, err :=

File: ddl/column.go [hunk 2]
```diff
@@ -271,22 +258,14 @@ func (d *ddl) onDropColumn(t *meta.Meta, job *model.Job) error {
...
```
Comment (ngaut) [inline ddl/column.go]: reorgInfo, err :=

File: ddl/index.go [hunk 3]
```diff
@@ -160,22 +159,14 @@ func (d *ddl) onCreateIndex(t *meta.Meta, job *model.Job) error {
...
-		if job.SnapshotVer == 0 {
-			var ver kv.Version
-			ver, err = d.store.CurrentVersion()
-			if err != nil {
-				return errors.Trace(err)
-			}
-
-			job.SnapshotVer = ver.Ver
+		var reorgInfo *reorgInfo
+		reorgInfo, err = d.getReorgInfo(t, job)
```
Comment (ngaut) [inline ddl/index.go]: reorgInfo, err :=

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**